src/js/builtins/shell.ts

```diff
@@ -282,9 +282,9 @@ export function createBunShellTemplateFunction(ShellInterpreter) {
       throw new TypeError("Class constructor Shell cannot be invoked without 'new'");
     }
 
-    var Shell = function Shell() {
-      const [first, ...rest] = arguments;
-      if (first.raw === undefined) throw new Error("Please use `$` as a tagged template function: $`cmd arg1 arg2`");
+    var Shell = function Shell(...args) {
```

Comment:
...args causes it to clone the arguments array

and then later, ...rest clones it again

we should use `arguments` more directly or alternatively we could do this destructuring in the function definition like (first, ...rest)

Reply:
`arguments` directly is a likely slower route. The `arguments` object has a lot of baggage with it and potential slowness if touched wrong. WebKit has magic `@` helpers for it though. However, in this case the `first, ...rest` is very nice!

Reply:
arguments is a bytecode intrinsic, it won’t materialize an extra object unless necessary  

… means it must always clone 

really the faster way here would be to pass through arguments without the … and then make it an ArgList and reference that. 

You can read JSC code to learn more about this if you search for ClonedArguments

Reply:
> it won’t materialize an extra object unless necessary

That maps to my comment "and potential slowness if touched wrong". Back with my time as the Chakra JS engine performance technical PM at Microsoft we had several gaps in our `arguments` object handling that would slow path us (compared to the competition) which we eventually fixed. Since then I have a weariness of `arguments` (even knowing other engines optimize it in more cases).

I'm speaking generally as maps to my "is a likely slower route" comment. And generally having devs reach for `arguments` should be avoided. Yes, if handled with care it can be okay. But JSC itself "has magic `@` helpers":
https://github.com/WebKit/WebKit/commit/61208896b94b2745450e56a744669977ff1dc0b1#diff-2f6c0607ce64738a74242e2c1419a66b1f9e622f40f36be549f41b4be3cf0fd4

so that they can, in their own JS implemented methods, avoid `arguments`.

As for the `rest` param. While knowing the arguments up front and passing them without needing an array is best, likely followed by magic helper use, of the syntax sugars `rest`, unlike spread, has a secure JS benefit as `rest` will not defer to `Array.prototype[Symbol.iterator]` and so is safe for internal use. The `rest` syntax also has DFG optimizations around it: 
* https://github.com/WebKit/WebKit/commit/babbd952c15c1948a7b1e6a4d63d9d7eb2bb2ae8
* https://github.com/WebKit/WebKit/commit/1908569d19e07989d8d21671723e0c9a5dd1449c
* [WebKit/WebKit@`329af67`](https://github.com/WebKit/WebKit/commit/329af672df18d52add0089e20121c4ac09add9ff#diff-91a5b64badfd992fd984615a03f88c669ac07d10d7671f5052a2fed25335052dR35) _(`rest` participates in the DFGArgumentsEliminationPhase too)_